partiql-parser/src/main/kotlin/org/partiql/parser/PartiQLParser.kt

```diff
@@ -14,19 +14,39 @@
 
 package org.partiql.parser
 
+import org.partiql.ast.Expr
 import org.partiql.ast.Statement
 import org.partiql.parser.internal.PartiQLParserDefault
+import org.partiql.spi.errors.PErrorListenerException
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.nullValue
+import kotlin.jvm.Throws
 
 public interface PartiQLParser {
 
-    @Throws(PartiQLSyntaxException::class, InterruptedException::class)
-    public fun parse(source: String): Result
+    /**
+     * Parses the [source] into an AST.
+     * @param source the user's input
+     * @param config a configuration object for the parser
+     * @throws PErrorListenerException when the [org.partiql.spi.errors.PErrorListener] defined in the [config] throws an
+     * [PErrorListenerException], this method halts execution and propagates the exception.
+     */
+    @Throws(PErrorListenerException::class)
+    public fun parse(source: String, config: ParserConfig = ParserConfigBuilder().build()): Result
 
     public data class Result(
         val source: String,
         val root: Statement,
         val locations: SourceLocations,
-    )
+    ) {
+        public companion object {
+            @OptIn(PartiQLValueExperimental::class)
+            internal fun empty(source: String): Result {
+                val locations = SourceLocations.Mutable().toMap()
+                return Result(source, Statement.Query(Expr.Lit(nullValue())), locations)
+            }
+        }
+    }
```

Comment:
We can remove the data class. I'm also curious your thoughts on these "Result" classes and if we should be pushing for better consistency across the components.

Reply:
Discussed. I'm in favor, though outside of this PR.